caas/kubernetes/provider/bootstrap.go

```diff
@@ -1104,15 +1107,23 @@ func (c *controllerStack) buildStorageSpecForController(statefulset *apps.Statef
 		},
 	})
 
-	fileMode := int32(256)
-	var vols []core.Volume
-	// add volume server.pem secret.
-	vols = append(vols, core.Volume{
+	vols := []core.Volume{{
+		Name: mongoScratchStorageName,
+		VolumeSource: core.VolumeSource{
+			EmptyDir: &core.EmptyDirVolumeSource{},
+		},
+	}, {
+		Name: apiServerScratchStorageName,
+		VolumeSource: core.VolumeSource{
+			EmptyDir: &core.EmptyDirVolumeSource{},
+		},
+	}, {
+		// add volume server.pem secret.
 		Name: c.resourceNameVolSSLKey,
 		VolumeSource: core.VolumeSource{
 			Secret: &core.SecretVolumeSource{
 				SecretName:  c.resourceNameSecret,
-				DefaultMode: &fileMode,
+				DefaultMode: pointer.Int32(0400),
```

Comment:
Small knit but I hate magic numbers. Can we define this with a name?

Reply:
Its a file mode. I don't think this needs any more.